filebrowser_safe/views.py

```diff
@@ -173,30 +180,47 @@ def browse(request):
                     results_var["select_total"] += 1
             except OSError:
                 # Ignore items that have problems
-                continue
+                return
             else:
-                files.append(fileobject)
+                sort_values[fileobject.filename] = getattr(fileobject, query["o"])
+                files_to_sort.append(fileobject)
                 results_var["results_current"] += 1
 
-        # COUNTER/RESULTS
-        if fileobject.filetype:
-            counter[fileobject.filetype] += 1
+    for i in range(0, file_list_size, batch_size):
+        file_list = file_list[i : i + batch_size]
+        for file in file_list:
+            # EXCLUDE FILES MATCHING ANY OF THE EXCLUDE PATTERNS
+            filtered = not file or file.startswith(".")
+            for re_prefix in filter_re:
+                if re_prefix.search(file):
+                    filtered = True
+            if filtered:
+                continue
+            proc_file = threading.Thread(
```

Comment:
What confines the number of threads in play? Anything other than I/O? Is that enough?

Reply:
I batch these calls so the configurable batch size would limit i/o, but because this is pretty much i/o bound, these threads should be pretty cheap. I can lower the batch size later if we need later.